onnxruntime/core/framework/allocator.cc

```diff
@@ -43,6 +43,7 @@ bool IAllocator::CalcMemSizeForArrayWithAlignment(size_t nmemb, size_t size, siz
 void* AllocatorDefaultAlloc(size_t size) {
   const size_t alignment = MlasGetPreferredBufferAlignment();
   if (size <= 0) return nullptr;
+  size += MLAS_SYMM_QGEMM_BUF_OVERRUN;
   void* p;
```

Comment:
Why do we add a constant overrun amount vs. rounding up the size to the required boundary?

Reply:
Rounding up is not enough.  For instance, the activation tensor is 8 by 8. So size is 64, alignment is good. But the last row only has 8 values.  The kernel needs to have a vector register to load the last row. It takes 16 byte to fill that vector register. Thus the kernel still overread by 8 bytes.

Reply:
Sorry - not quite following. If the last row is 8 and you need a multiple of 16, can't you determine that only 8 additional are required? Adding 15 is obviously simpler - was just wondering if there was a reason other than that to not take into account the actual number needed.